kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/AddFile.java

```diff
@@ -55,12 +58,15 @@ public class AddFile {
           .add(
               "tags",
               new MapType(StringType.STRING, StringType.STRING, true /* valueContainsNull */),
-              true /* nullable */);
+              true /* nullable */)
+          .add("baseRowId", LongType.LONG, true /* nullable */)
+          .add("defaultRowCommitVersion", LongType.LONG, true /* nullable */);
 
   public static final StructType SCHEMA_WITH_STATS = SCHEMA_WITHOUT_STATS.add(JSON_STATS_FIELD);
 
   /** Full schema of the {@code add} action in the Delta Log. */
   public static final StructType FULL_SCHEMA = SCHEMA_WITH_STATS;
+
```

Comment:
Can't leave a comment on line 81 so putting it here:
```
  public static Row convertDataFileStatus(
```
This is a good example of what I mentioned in the slack thread:
> - All public kernel APIs should use Row as this is what engines understand
- All internal kernel APIs should use concrete, internal types such as AddFile
- ~(Optional?)~ We should convert from Row to concrete type at the boundary with engines so that all the internal kernel code doesn't need to use Row

This is an internal API, so it should return an `AddFile`.
It's then used in `generateAppendActions` which is a public API, which is where we'll want to turn into a row:
```
  static CloseableIterator<Row> generateAppendActions(...) {
          AddFile addFile =
              AddFile.convertDataFileStatus(
                  tableRoot,
                  dataFileStatus,
                  ((DataWriteContextImpl) dataWriteContext).getPartitionValues(),
                  true /* dataChange */);
          return SingleAction.createAddFileSingleAction(addFile.toRow());
```